tests/DataCache.test.ts

```diff
@@ -0,0 +1,136 @@
+import { describe, expect, it } from "vitest";
+
+import { ANY_ERROR } from "./test_utils";
+
+import DataCache from "../src/colorizer/FrameCache";
+
+describe("DataCache", () => {
+  class DisposableString {
+    public value: string;
+    public hasBeenDisposed: boolean = false;
+
+    constructor(value: string) {
+      this.value = value;
+    }
+
+    dispose(): void {
+      this.hasBeenDisposed = true;
```

Comment:
it would also be interesting to test caching a type that doesn't have a dispose method, like a simple Array or ArrayBuffer.

Reply:
Added!